examples/hosts/app-integration/schema-upgrade/src/app.tsx

```diff
@@ -0,0 +1,153 @@
+// we'll choose to use the current timestamp.  We'll also choose to interpret the URL hash as an existing document's
+// ID to load from, so the URL for a document load will look something like http://localhost:8080/#1596520748752.
+// These policy choices are arbitrary for demo purposes, and can be changed however you'd like.
+let createNew = false;
+if (location.hash.length === 0) {
+    createNew = true;
+    location.hash = Date.now().toString();
+}
+const documentId = location.hash.substring(1);
+document.title = documentId;
+
+async function getInventoryListFromContainer(container: IContainer): Promise<IInventoryList> {
+    // Since we're using a ContainerRuntimeFactoryWithDefaultDataStore, our inventory list is available at the URL "/".
+    const url = "/";
+    const response = await container.request({ url });
+
+    // Verify the response to make sure we got what we expected.
+    if (response.status !== 200 || response.mimeType !== "fluid/object") {
+        throw new Error(`Unable to retrieve data object at URL: "${url}"`);
+    } else if (response.value === undefined) {
+        throw new Error(`Empty response from URL: "${url}"`);
+    }
+
+    return response.value as IInventoryList;
+}
+
+async function getContainerKillBitFromContainer(container: IContainer): Promise<IContainerKillBit> {
+    // Since we're using a ContainerRuntimeFactoryWithDefaultDataStore, our inventory list is available at the URL "/".
+    const url = containerKillBitId;
+    const response = await container.request({ url });
+
+    // Verify the response to make sure we got what we expected.
+    if (response.status !== 200 || response.mimeType !== "fluid/object") {
+        throw new Error(`Unable to retrieve data object at URL: "${url}"`);
+    } else if (response.value === undefined) {
+        throw new Error(`Empty response from URL: "${url}"`);
+    }
+
+    return response.value as IContainerKillBit;
+}
+
+async function start(): Promise<void> {
+    const tinyliciousService = new TinyliciousService();
+
+    const module = { fluidExport: new InventoryListContainerRuntimeFactory() };
+    const codeLoader = { load: async () => module };
+
+    const loader = new Loader({
+        urlResolver: tinyliciousService.urlResolver,
+        documentServiceFactory: tinyliciousService.documentServiceFactory,
+        codeLoader,
+    });
+
+    let fetchedData: string | undefined;
+    let container: IContainer;
+    let inventoryList: IInventoryList;
+    let containerKillBit: IContainerKillBit;
+
+    if (createNew) {
+        fetchedData = await fetchData();
+        container = await loader.createDetachedContainer({ package: "no-dynamic-package", config: {} });
+        inventoryList = await getInventoryListFromContainer(container);
+        containerKillBit = await getContainerKillBitFromContainer(container);
+        await applyStringData(inventoryList, fetchedData);
```

Comment:
The idea of the pattern here is to target **new** container with older serialized data? I'm assuming you could target original container as well.

Reply:
Possibly - at the moment, a new container is the clearest path forward but I expect we'll also want to explore upgrade in place.